using_images/xpaas_images/jws.adoc

```diff
@@ -9,7 +9,64 @@
+
+[IMPORTANT]
+There are significant differences in the functionality between the JBoss Web Server xPaaS images and the regular release of JBoss Web Server.
+
+This topic details the differences between the JBoss Web Server xPaaS images and the regular release of JBoss Web Server, and provides instructions specific to running and configuring the JBoss Web Server xPaaS images. Documentation for other JBoss Web Server functionality not specific to the JBoss Web Server xPaaS images can be found in the https://access.redhat.com/documentation/en-US/Red_Hat_JBoss_Web_Server/[JBoss Web Server documentation on the Red Hat Customer Portal].
+
+The location of `_JWS_HOME_/tomcat<version>/` inside a JBoss Web Server xPaaS image is: *_/opt/webserver/_*.
+
+== Functionality Differences in the OpenShift JBoss Web Server xPaaS Images
+
+A major functionality difference compared to the regular release of JBoss Web Server is that there is no Apache HTTP Server in the OpenShift JBoss Web Server xPaaS images. All load balancing in OpenShift is handled by the OpenShift router, so there is no need for a load-balancing Apache HTTP Server with mod_cluster or mod_jk connectors. 
+
+== Installing the JBoss Web Server xPaaS Image Streams and Application Templates
+
+To use the Red Hat xPaaS middleware images in your OpenShift project, you must first install the image streams and Source-to-Image (S2I) application templates.
+
+. Clone the Application Template repository:
++
+----
+# git clone https://github.com/jboss-openshift/application-templates
+----
+. Create the image streams and S2I templates:
++
+----
+# oc create -f application-templates/jboss-image-streams.json -n openshift
+# oc create -f application-templates/webserver/ *.json -n openshift
+----
+
+The JBoss Web Server xPaaS images streams and S2I application templates are now available to use in your OpenShift project.
+
+== Using the JBoss Web Server xPaaS Image Source-to-Image (S2I) Process
+
+To run and configure the OpenShift JBoss Web Server xPaaS images, use the OpenShift S2I process with the application template parameters and environment variables.
+
+The S2I process for the JBoss Web Server xPaaS images works as follows:
+
+. If there is a *_pom.xml_* file in the source repository, a Maven build is triggered with the contents of `*$MAVEN_ARGS*` environment variable.
++
+By default the `package` goal is used with the `openshift` profile, including the system properties for skipping tests (`*-DskipTests*`) and enabling the Red Hat GA repository (`*-Dcom.redhat.xpaas.repo.redhatga*`).
++
+The results of a successful Maven build are copied to *_/opt/webserver/webapps_*. By default, this includes all WAR files from the *_target_* directory in the source repository.
+. All WAR files from the directory within the source repository specified by the `*$ARTIFACT_DIR*` environment variable are copied to *_/opt/webserver/webapps_*.
```

Comment:
Also I think this is confused.  The previous bullet and this bullet are saying the same thing, because the default value is "target" and if you override the environment variable value, then nothing is copied out of "target":
http://git.app.eng.bos.redhat.com/git/jboss-dockerfiles.git/tree/openshift/webserver/tomcat8/3.0/scripts/added/sti/assemble#n44
(My understanding is it's a bug that it uses "target" and not "$ARTIFACT_DIR" currently, but hopefully you get my point)

What's missing is an additional step which is to copy the contents of deployments, regardless of whether a pom.xml is present.  
http://git.app.eng.bos.redhat.com/git/jboss-dockerfiles.git/tree/openshift/webserver/tomcat8/3.0/scripts/added/sti/assemble#n57